core/serialize/serialize.cc

```diff
@@ -904,6 +904,7 @@ void SerializerImpl::pickle(Pickler &p, const ast::TreePtr &what) {
             // Can replace this with std::bit_cast in C++20
             memcpy(&flags, &s->flags, sizeof(flags));
             p.putU1(flags);
+            p.putU4(s->numPosArgs);
```

Comment:
This is serialized as a u4 because we didn't have a `putU2` function available. If there's a spec for the format, I'd be happy to add support for u2 as well, but given that it seems to be doing leading zero compression, it didn't seem too bad to use a u4 here.